spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientConfigMapPropertySource.java

```diff
@@ -46,17 +47,18 @@ public KubernetesClientConfigMapPropertySource(CoreV1Api coreV1Api, String name,
 			Environment environment) {
 
 		try {
-			List<String> names = new ArrayList<>();
+			Set<String> names = new HashSet<>();
 			names.add(name);
 			if (environment != null) {
 				for (String activeProfile : environment.getActiveProfiles()) {
 					names.add(name + "-" + activeProfile);
 				}
 			}
-			Map<String, Object> result = new LinkedHashMap<>();
+			Map<String, Object> result = new HashMap<>();
```

Comment:
the order of properties that are stored inside a config map should not matter (we did the same change in fabric8 btw), so replace with a `HashMap`

Reply:
Hmmm it could matter though.

If (for some strange reason) you have the same property defined twice in a config map I would expect the last one to take precedence.

Reply:
I am not very sure I understand your point here; this code is not about same property from same config map. In contrast, it is about properties from different config maps.

So you process `config-map-A`, get all of its properties (the order in which you get them is undefined for both clients), put these into `result`. Process `config-map-B`, put all its properties into same `result`. As such, "last property" wins, is still preserved. If I understood your point correctly.

Reply:
Makes sense